### PR TITLE
added With node

### PR DESCRIPTION
**Pull request by nivanov-ati**

### CODE REVIEW

File: common/decorator/repeat_until.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ func (d *repeatUntil[Blackboard]) Tick(ctx context.Context, bb Blackboard, evt c
 	log.Info().Msg("Repeater: Calling child")
 	status := core.Update(ctx, d.Child, bb, evt)
 
-	if status == core.StatusError || status == core.StatusInvalid {
+	if status.Status() == core.StatusError || status.Status() == core.StatusInvalid {
```
Comment (jbcpollak) [inline common/decorator/repeat_until.go:45]: I wonder if we should be checking for the `err` in StatusError

**Review by jbcpollak — APPROVED** [review]
I think especially for this new "interesting" decorator you should write a quick unit test to demonstrate it and make sure it works.

File: common/decorator/repeat_until.go (hunk 1 above)
Comment (nivanov-ati) [reply common/decorator/repeat_until.go:45]: I dont think we should check it here - whether we have a StatusError or a NodeRuntimeError (which are both StatusError-like) we want to return that. However, greenstalk only panics on NodeRuntimeError and StatusError doesn't really do much, I don't think. We may want to consider merging StatusError and NodeRuntimeError

Comment (jbcpollak) [reply common/decorator/repeat_until.go:45]: sorry - yes I was thinking this was in greenstalk.go - we should merge StatusError and NodeRuntimeError and have greenstalk.go handle it there.

**Comment by nivanov-ati** [comment]
@jbcpollak added a test. Also I tested the node directly in firefly by pulling in the new version from this branch

**Review by jbcpollak — APPROVED** [review]
LGTM

**Pull request merged**